processing/src/main/java/io/druid/query/PrioritizedExecutorService.java

```diff
@@ -107,9 +115,8 @@ public PrioritizedExecutorService(
     );
     return PrioritizedListenableFutureTask.create(
         ListenableFutureTask.create(runnable, value),
-        runnable instanceof PrioritizedRunnable
-        ? ((PrioritizedRunnable) runnable).getPriority()
-        : defaultPriority
+        getPriority(runnable),
```

Comment:
not sure if extracting this method out makes much of a difference, if anything I find it confusing to have getPriority take an Object.